pypeit/spectrographs/vlt_xshooter.py

```diff
@@ -624,6 +571,10 @@ def get_detector_par(self, det, hdu=None):
         # Binning
         # TODO: Could this be detector dependent??
         binning = '1,1' if hdu is None else self.get_meta_value(self.get_headarr(hdu), 'binning')
+        
+        # Grab the gain and read noise from the header.
+        gain = None if hdu is None else hdu[0].header['HIERARCH ESO DET OUT1 CONAD']
```

Comment:
Note sure how the code will react to a `gain=None` (also `ronoise = None`). Note that `hdu` is an optional parameter, so it can be None.
Can you just use a default value (maybe the one that was used before)?

Reply:
I borrowed the `None`s from from the KCWI implementation. I think the idea is that if there is no hdu, then no image is actually being processed (e.g. during sensfunc or coadd1d), so those values are irrelevant. But I can put in the nominal values if it might otherwise be necessary.

Reply:
Actually, the main reason these functions to allow `hdu=None` is so that [this table](https://pypeit.readthedocs.io/en/release/detectors.html#instrument-specific-data) can be built automatically.  Entries in the table set to None mean that the data are read from the files being processed.